docs/spec-files/problems.md

```diff
@@ -0,0 +1,88 @@
+
+In addition, messages MAY use the [Advanced Sequencing](../../extensions/advanced_sequencing/main.md) extension to detect gaps in delivery or messages arriving out of order.
+
+### Explicit problem reports
+
+DIDComm features a standard method for reporting problems (errors) to other parties. This is not always possible (e.g., when a sender has o route to the other party, or when a recipient's crypto is incompatible with a sender's). But when it is possible, it can be very helpful.
+
+>NOTE: DIDComm calls these things "problems" instead of "errors" because we use the same reporting mechanism for both errors and warnings.
+
+The mechanism requires the reporting party to send a simple message called a **problem report** that looks like this:
+
+```json
+{
+  "type": "https://didcomm.org/notify/1.0/problem-report",
+  "id": "7c9de639-c51c-4d60-ab95-103fa613c805",
+  "pthid": "1e513ad4-48c9-444e-9e7e-5b8b45c5e325",
+  "body": {
+    "code": "cant-find-route",
+    "comment": "Unable to find a route to the specified recipient.",
+    "args": [
+      "did:sov:C805sNYhMrjHiqZDTUASHg"
+    ],
+    "cause": "",
+    "trigger": "did:example:foo",
+    "state_outcome": null,
+    "escalate_to": "mailto:admin@foo.org"
+  }
+}
+```
+
+The `pthid` header MUST be included with problem reports. It should contain the `thid` of the thread in which the problem occurred. (Thus, the problem report begins a new child thread, of which the triggering context is the parent.)
+
+The required `code` field is a machine-readable string -- lower kabob case by convention -- that identifies the problem. Codes are formally defined in and namespaced by the protocol definitions where they are triggered -- so in the example above, the protocol that was underway in thread `1e513ad4-48c9-444e-9e7e-5b8b45c5e325` defined a code, `cant-find-route` that was used in this problem report.
+
+The optional `comment` field contains localizable, human-friendly text.
+
+The optional `args` field contains args that are associated with the code when it is defined. In this example, `args` apparently lists a DID that cannot be routed.
+
+The optional `cause` field allows the sender to include a stack trace or other background material to explain what triggered the error. This is akin to the `cause` property of Java's `Throwable`. 
+
+The optional `owner` field declares who the reporter of the problem considers to have the responsibility to do something about it. Typically this would be one of the parties in the protocol. If the sender plans to retry, it would be the sender's DID; if the sender wants to recipient to fix the problem, it would be the recipient's DID.
+
+The required `state_outcome` field tells what the sender's state state machine is in, in the parent protocol, after processing this error. In this case, the sender's state has reverted to `null`, meaning it considers the protocol to be aborted.
```

Comment:
Let's make sure to discuss this. I'm concerned that revealing information about the state machine could be used as a method to reverse engineer or conduct reconnaissance on an implementation. It's not a hard no for me to not include this, but I'm skeptical about over revealing error states which is why I went the error code route in #83